assembly/src/assembler/instruction/field_ops.rs

```diff
@@ -95,22 +95,89 @@ pub fn pow2(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError>
     Ok(None)
 }
 
+/// Computes `ilog2` for the number at the top of the stack, and its corresponding power-of-two.
+///
+/// Input: [n, ...]
+/// Output: [pow2, ilog2, ...]
```

Comment:
The instruction is return the `pow2` because that is computed as part of the validation of the advice stack value, and it is a useful value to expose (for example the MMR use that, and this saves 16 cycles there).

But that means the name of the instruction isn't great, I couldn't think of a better alternative :(